### PR TITLE
drivers: display: st7789v: convert to MIPI DBI API

### PR DESCRIPTION
**Pull request by danieldegrasse**
Convert ST7789V display driver to use MIPI DBI API. Update in tree boards to use the new devicetree syntax needed to enable
this display with the MIPI DBI API.

### CODE REVIEW

File: doc/releases/migration-guide-3.7.rst [hunk 1]
```diff
@@ -317,6 +317,50 @@ Display
         };
     };
 
+
+* ST7789V based displays now use the MIPI DBI driver class. These displays
+  must now be declared within a MIPI DBI driver wrapper device, which will
+  manage interfacing with the display. Note that the `cmd-data-gpios` pin has
+  changed polarity with this update, to align better with the new
+  `dc-gpios` name. For an example, see below:
```
Comment (aescolar) [inline doc/releases/migration-guide-3.7.rst:325]:
```suggestion
  `dc-gpios` name. (:github:`73750`) For an example, see below:
```

Comment (danieldegrasse) [reply doc/releases/migration-guide-3.7.rst:325]: Done, thanks

**Review by jfischer-no — CHANGES_REQUESTED** [review]
It does not work e.g. with st7789v_waveshare_240x240 on nRF52840DK (or FRDM-K64), while main works as expected.

`spi_nrfx_spim: Word sizes other than 8 bits are not supported`

**Comment by jfischer-no** [comment]
> > It does not work e.g. with st7789v_waveshare_240x240 on nRF52840DK (or FRDM-K64), while main works as expected.
> > `spi_nrfx_spim: Word sizes other than 8 bits are not supported`
> 
> Thank you for testing this- I checked, and the SPI word size was being set incorrectly- this change should fix it, just pushed an update:
> 
> ```diff
> diff --git a/drivers/display/display_st7789v.c b/drivers/display/display_st7789v.c
> index ce2cd2bcd63..1f7b970b126 100644
> --- a/drivers/display/display_st7789v.c
> +++ b/drivers/display/display_st7789v.c
> @@ -365,8 +365,8 @@ static const struct display_driver_api st7789v_api = {
>  #define ST7789V_INIT(inst)                                                             \
>         static const struct st7789v_config st7789v_config_ ## inst = {                  \
>                 .mipi_dbi = DEVICE_DT_GET(DT_INST_PARENT(inst)),                        \
> -               .dbi_config = MIPI_DBI_CONFIG_DT_INST(inst, SPI_WORD_SET(               \
> -                                                     ST7789V_WORD_SIZE(inst)) |        \
> +               .dbi_config = MIPI_DBI_CONFIG_DT_INST(inst,                             \
> +                                                     ST7789V_WORD_SIZE(inst) |         \
>                                                       SPI_OP_MODE_MASTER, 0),           \
>                 .vcom = DT_INST_PROP(inst, vcom),                                       \
>                 .gctrl = DT_INST_PROP(inst, gctrl),                                     \
> ```

Yes, I noticed that, but there is also something wrong with data transfer, it looks like random data on the screen (samples/drivers/display).

**Comment by danieldegrasse** [comment]
> Yes, I noticed that, but there is also something wrong with data transfer, it looks like random data on the screen (samples/drivers/display).

I believe we were missing the `ST7789V_CMD_RAMWR` command in the `display_write` implementation (a bit critical, I think). Thanks again for testing this out, sorry for the churn

**Review by jfischer-no — APPROVED** [review]
LGTM, tested with st7789v_waveshare_240x240 (samples/drivers/display) on nRF52840DK and FRDM-K64F.

**Review by dleach02 — APPROVED** [review]

**Pull request merged**